### PR TITLE
Offline Request Queue

### PR DESCRIPTION
**Pull request by summep**

### CODE REVIEW

File: src/Form/index.tsx [hunk 1]
```diff
@@ -433,6 +434,23 @@ function Form({
     }
   }, [stepKey]);
 
+  useEffect(() => {
```
Comment (bo-dun-1) [inline src/Form/index.tsx:437]: this useEffect coudl maybe be moved into the offlineRequestHandler file as well, and a hook can be returned from the file. e.g. `useOfflineRequestHandler()`

File: src/utils/featheryClient/index.ts [hunk 2]
```diff
@@ -140,12 +156,27 @@ export default class FeatheryClient extends IntegrationClient {
...
       body: formData,
       // In Safari, request fails with keepalive = true if over 64kb payload.
       keepalive: false
-    });
+    };
+    if (navigator.onLine) {
+      return this._fetch(url, options);
+    } else {
+      const { sdkKey } = initInfo();
+      options = {
```
Comment (bo-dun-1) [inline src/utils/featheryClient/index.ts:170]: options.headers.Authorization = 'Token ' + sdkKey;

File: src/utils/featheryClient/index.ts [hunk 3]
```diff
@@ -441,7 +472,18 @@ export default class FeatheryClient extends IntegrationClient {
     }
     if (userId) formData.set('fuser_key', userId);
 
-    return this._fetch(url, { method: 'POST', body: formData });
+    if (navigator.onLine) {
```
Comment (bo-dun-1) [inline src/utils/featheryClient/index.ts:475]: const options = { method: 'POST', body: formData };
if (navigator.onLine) return this._fetch(url, options);
else {
  const request = new Request(...
}

File: src/utils/featheryClient/index.ts [hunk 4]
```diff
@@ -487,19 +529,33 @@ export default class FeatheryClient extends IntegrationClient {
...
+          this.draft ? Promise.resolve() : this._fetch(url, options)
+        )
+      );
+    } else {
+      const { sdkKey } = initInfo();
+      options = {
+        ...options,
+        headers: {
+          ...options.headers,
+          Authorization: 'Token ' + sdkKey
```
Comment (bo-dun-1) [inline src/utils/featheryClient/index.ts:551]: options.headers.Authorization = ...

File: src/utils/featheryClient/index.ts [hunk 5]
```diff
@@ -86,14 +89,27 @@ export default class FeatheryClient extends IntegrationClient {
...
-    return this._fetch(url, options);
+    if (navigator.onLine) {
+      return this._fetch(url, options);
+    } else {
+      const { sdkKey } = initInfo();
+      options = {
+        ...options,
+        headers: {
+          ...options.headers,
+          Authorization: 'Token ' + sdkKey
```
[the remark this answers is not in the document]
Comment (summep) [reply src/utils/featheryClient/index.ts:107]: Typescript does not like it as options.header can be undefined, so changed to:
      options.headers = {
        ...options.headers,
        Authorization: `Token ${sdkKey}`
      };

**Pull request closed without merging**